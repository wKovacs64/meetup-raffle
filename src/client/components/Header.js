import React from 'react';
import GitHubCorner from 'react-github-corner';

const Header = () => (
  <header className="ph3 pv3 bg-light-red">
    <GitHubCorner
      href="https://github.com/wKovacs64/meetup-raffle"
      octoColor="#f4f4f4"
      bannerColor="#111"
    />
    <div className="dt vh-25-ns center-ns">
      <div className="dtc v-mid">
        <h1 className="f3 f2-m f1-l ma0 lh-title near-white">
          <span role="img" aria-hidden>
            🤝
          </span>{' '}
          meetup-raffle{' '}
          <span role="img" aria-hidden>
            🎟️
          </span>
        </h1>
        <h2 className="f6 f4-m f3-l ma0 lh-title">
          Draw raffle winners at your Meetup event
        </h2>
      </div>
    </div>
  </header>
);

Header.displayName = 'Header';

export default Header;
