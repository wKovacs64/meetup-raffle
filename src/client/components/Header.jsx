/** @jsx jsx */
import { jsx, useThemeUI, Box, Flex, Heading } from 'theme-ui';
import GitHubCorner from 'react-github-corner';

function Header() {
  const { theme } = useThemeUI();

  return (
    <header sx={{ bg: 'accent', p: 3 }}>
      <GitHubCorner
        bannerColor={theme.colors.text}
        href="https://github.com/wKovacs64/meetup-raffle"
        target="_blank"
        rel="noreferrer noopener"
      />
      <Flex
        sx={{
          height: ['auto', '25vh'],
          justifyContent: ['normal', 'center'],
          alignItems: 'center',
        }}
      >
        <Box>
          <Heading
            as="h1"
            variant="text.title"
            sx={{
              fontSize: [4, 5, 6],
              textAlign: ['left', 'center'],
            }}
          >
            <span role="img" aria-hidden>
              🤝
            </span>{' '}
            meetup-raffle{' '}
            <span role="img" aria-hidden>
              🎟️
            </span>
          </Heading>
          <Heading as="h2" variant="text.subtitle" sx={{ fontSize: [1, 3, 4] }}>
            Draw raffle winners at your Meetup event
          </Heading>
        </Box>
      </Flex>
    </header>
  );
}

Header.displayName = 'Header';

export default Header;
