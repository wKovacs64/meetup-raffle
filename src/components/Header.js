import React from 'react';
import GitHubCorner from 'react-github-corner';

export default () => (
  <header className="ph3 pv3 bg-light-red">
    <GitHubCorner
      href="https://github.com/wKovacs64/meetup-raffle-web"
      octoColor="#f4f4f4"
      bannerColor="#111"
    />
    <div className="dt vh-25-ns center-ns">
      <div className="dtc v-mid">
        <h1 className="f3 f2-m f1-l ma0 lh-title near-white">
          <span role="img" aria-label="meeting">
            🤝
          </span>{' '}
          meetup-raffle{' '}
          <span role="img" aria-label="ticket">
            🎟️
          </span>
        </h1>
        <h3 className="f6 f4-m f3-l ma0 lh-title">
          Draw raffle winners at your Meetup event
        </h3>
      </div>
    </div>
  </header>
);
