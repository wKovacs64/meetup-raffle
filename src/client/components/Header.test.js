import React from 'react';
import { render } from 'react-testing-library';
import Header from './Header';

describe('Header', () => {
  const { container } = render(<Header />);

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });
});
