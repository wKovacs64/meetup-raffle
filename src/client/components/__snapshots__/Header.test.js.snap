// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <header
    class="ph3 pv3 bg-light-red"
  >
    <a
      href="https://github.com/wKovacs64/meetup-raffle-web"
      class="github-corner"
      aria-label="Open GitHub project"
    >
      <svg
        width="80"
        height="80"
        viewBox="0 0 250 250"
        style="position: absolute; top: 0px; fill: #f4f4f4; right: 0px;"
      >
        <path
          class="octo-banner"
          d="M0 0l115 115h15l12 27 108 108V0z"
          fill="#111"
        />
        <path
          class="octo-arm"
          d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
        />
        <path
          class="octo-body"
          d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
        />
      </svg>
    </a>
    <div
      class="dt vh-25-ns center-ns"
    >
      <div
        class="dtc v-mid"
      >
        <h1
          class="f3 f2-m f1-l ma0 lh-title near-white"
        >
          <span
            role="img"
            aria-label="meeting"
          >
            🤝
          </span>
          meetup-raffle
          <span
            role="img"
            aria-label="ticket"
          >
            🎟️
          </span>
        </h1>
        <h3
          class="f6 f4-m f3-l ma0 lh-title"
        >
          Draw raffle winners at your Meetup event
        </h3>
      </div>
    </div>
  </header>
</div>
`;
